gpjax/gps.py

```diff
@@ -106,64 +152,69 @@ class ConjugatePosterior(Posterior):
+        obs_noise = params["likelihood"]["obs_noise"]
+        mx = self.prior.mean_function(x, params["mean_function"])
+        
         # Precompute covariance matrices
-        Kff = gram(self.prior.kernel, X, params["kernel"])
-        prior_mean = self.prior.mean_function(X, params["mean_function"])
-        L = cho_factor(Kff + I(n_train) * sigma, lower=True)
+        Kxx = gram(self.prior.kernel, x, params["kernel"])
+        Kxx += I(nx) * self.jitter
+        Lx = cho_factor(Kxx + I(nx) * obs_noise, lower=True)
 
-        prior_distance = y - prior_mean
-        weights = cho_solve(L, prior_distance)
+        weights = cho_solve(Lx, y - mx)
 
         def mean_fn(test_inputs: Array) -> Array:
-            prior_mean_at_test_inputs = self.prior.mean_function(test_inputs, params["mean_function"])
-            Kfx = cross_covariance(self.prior.kernel, X, test_inputs, params["kernel"])
-            return prior_mean_at_test_inputs + jnp.dot(Kfx, weights)
+            t = test_inputs
+            mt = self.prior.mean_function(t, params["mean_function"])
+            Ktx = cross_covariance(self.prior.kernel, t, x, params["kernel"])
+            return mt + jnp.dot(Ktx, weights)
 
         return mean_fn
 
-    def variance(self, training_data: Dataset, params: dict) -> tp.Callable[[Array], Array]:
-        X = training_data.X
-        n_train = training_data.n
+    def variance(self, train_data: Dataset, params: dict) -> tp.Callable[[Array], Array]:
+        x, nx = train_data.X, train_data.n
         obs_noise = params["likelihood"]["obs_noise"]
-        n_train = training_data.n
-        Kff = gram(self.prior.kernel, X, params["kernel"])
-        Kff += I(n_train) * 1e-8
-        L = cho_factor(Kff + I(n_train) * obs_noise, lower=True)
+        Kxx = gram(self.prior.kernel, x, params["kernel"])
+        Kxx += I(nx) * self.jitter
+        Lx = cho_factor(Kxx + I(nx) * obs_noise, lower=True)
 
         def variance_fn(test_inputs: Array) -> Array:
-            Kfx = cross_covariance(self.prior.kernel, X, test_inputs, params["kernel"])
-            Kxx = gram(self.prior.kernel, test_inputs, params["kernel"])
-            latent_values = cho_solve(L, Kfx.T)
-            return Kxx - jnp.dot(Kfx, latent_values)
+            t = test_inputs
+            Ktx = cross_covariance(self.prior.kernel, t, x, params["kernel"])
+            Ktt = gram(self.prior.kernel, t, params["kernel"])
+            latent_values = cho_solve(Lx, Ktx.T)
+            return Ktt - jnp.dot(Ktx, latent_values)
 
         return variance_fn
 
     def marginal_log_likelihood(
         self,
-        training: Dataset,
+        train_data: Dataset,
         transformations: tp.Dict,
         priors: dict = None,
         static_params: dict = None,
         negative: bool = False,
     ) -> tp.Callable[[Dataset], Array]:
-        x, y = training.X, training.y
+        x, y, nx = train_data.X, train_data.y, train_data.n
```

Comment:
Can we unpack as `x`, `y` and `n` or `n_data` - `nx` could be confusing as it kinda implies that `x` and `y` might be of different length.